src/linter.js

```diff
@@ -64,37 +70,42 @@ export default async function linter(options, compiler) {
     'ESLintWebpackPlugin',
     async (compilation) => {
       if (warnings.length > 0) {
+        // @ts-ignore
         compilation.warnings.push(new ESLintError(formatter.format(warnings)));
         warnings = [];
       }
 
       if (errors.length > 0) {
+        // @ts-ignore
         compilation.errors.push(new ESLintError(formatter.format(errors)));
         errors = [];
       }
     }
   );
 
-  compiler.hooks.emit.tapPromise('ESLintWebpackPlugin', async (compilation) => {
-    const { outputReport } = options;
+  compiler.hooks.afterEmit.tapPromise(
```

Comment:
Ideally it should be: 

```
compiler.hooks.thisCompilation.tap((compilation) => { 
  const Compilation = compiler.webpack.Compilation;
  compilation.hooks.process.processAssets({ 
    name: 'name', 
    stage:  Compilation.PROCESS_ASSETS_STAGE_REPORT}, () => {
      // Logic
    }) 
})`
```

https://github.com/webpack/webpack/blob/master/lib/Compilation.js#L3470

for wepback@4 we can use https://github.com/webpack/webpack/blob/master/lib/Compilation.js#L514, 

It allows other plugins to use report file for other purposes (rare case, but it is good to support this)

Reply:
is there an example I can reference?

Reply:
I've attempted to split concerns some so it would be easier to switch methods of emitting the report. However, the webpack api is new to me. I'm trying to implement your suggestions, but I'm fairly certain I am doing it wrong.

Reply:
@evilebottnawi `compilation.hooks.process` is undefined in webpack 5. `compilation.hooks.processAssets` is not a function. What am I missing? `compilation.hooks.processAssets.tap()`?  Stage doesn't seem to be honored?